compiler/rustc_typeck/src/coherence/orphan.rs

```diff
@@ -243,13 +245,25 @@ fn emit_orphan_check_error<'tcx>(
                     _ => ty,
                 };
                 let this = "this".to_string();
-                let (ty, postfix) = match &ty.kind() {
+                let (ty_str, postfix) = match &ty.kind() {
                     ty::Slice(_) => (this, " because slices are always foreign"),
```

Comment:
wait, why not handle `ty::RawPtr` here? like "because raw pointers are always foreign"?

Reply:
This was my original solution, but after that, I did not find the error message expressive enough? but now with the actual review can we make this squash of code?

I have no strong opinion on this side

Reply:
Can you make this change? I think it will simplify the code and I don't think it's that different of a message.

Reply:
Additionally, instead of saying `"this"` you can just always print the type name if you think that would improve this message.